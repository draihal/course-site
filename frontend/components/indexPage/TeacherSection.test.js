import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import TestTeacherSection from './TeacherSection'

const teachers = [
  {
    "teacher_id": 1,
    "url": "/api/v1/profile/teacher/1/",
    "avatar": null,
    "teacher_first_name": "Teacher 1",
    "teacher_last_name": "Test 1",
    "company": "",
    "position": "",
    "bio": ""
  },
  {
    "teacher_id": 2,
    "url": "/api/v1/profile/teacher/2/",
    "avatar":  null,
    "teacher_first_name": "Teacher 2",
    "teacher_last_name": "Test 2",
    "company": "",
    "position": "",
    "bio": "Testing bio 2"
  },
  {
    "teacher_id": 3,
    "url": "/api/v1/profile/teacher/3/",
    "avatar": null,
    "teacher_first_name": "Teacher 3",
    "teacher_last_name": "Test 3",
    "company": "",
    "position": "",
    "bio": "Testing bio 3"
  }
];


describe('TeacherSection', () => {
  it('render correctly TeacherSection component', () => {
    const tree = renderer.create(<TestTeacherSection teachers={teachers} />).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render TeacherSection correctly with given props', () => {
    const { queryByText } = render(<TestTeacherSection teachers={teachers} />);
    expect(queryByText(/Teacher 1/i)).toBeTruthy();
    expect(queryByText(/Teacher 2/i)).toBeTruthy();
    expect(queryByText(/Teacher 3/i)).toBeTruthy();
  });

});