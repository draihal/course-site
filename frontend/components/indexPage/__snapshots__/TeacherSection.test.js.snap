// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherSection render correctly TeacherSection component 1`] = `
<div
  className="offset"
  id="teachers"
>
  <div
    className="fixed-background"
  >
    <div
      className="row dark"
    >
      <div
        className="container"
      >
        <div
          className="narrow text-center"
        >
          <div
            className="col-12"
          >
            <h3
              className="heading"
            >
              Преподаватели
            </h3>
            <div
              className="heading-underline"
            >
               
            </div>
          </div>
        </div>
        <div
          className="row text-center"
        >
          <div
            className="col-md-4 teachers"
          >
            <h3>
              Teacher 1
               
              Test 1
            </h3>
            <div
              className="feature"
            >
              <img
                src="static/default_user.jpg"
              />
              <p
                className="lead"
              >
                
              </p>
            </div>
          </div>
          <div
            className="col-md-4 teachers"
          >
            <h3>
              Teacher 2
               
              Test 2
            </h3>
            <div
              className="feature"
            >
              <img
                src="static/default_user.jpg"
              />
              <p
                className="lead"
              >
                
              </p>
            </div>
          </div>
          <div
            className="col-md-4 teachers"
          >
            <h3>
              Teacher 3
               
              Test 3
            </h3>
            <div
              className="feature"
            >
              <img
                src="static/default_user.jpg"
              />
              <p
                className="lead"
              >
                
              </p>
            </div>
          </div>
        </div>
        <div
          className="text-center"
        >
          <div
            className="col-12"
          >
            <a
              className="btn btn-outline-light btn-lg"
              href="#"
            >
              Все преподаватели
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="fixed-wrap"
    >
      <div
        className="fixed"
      />
    </div>
  </div>
</div>
`;
